File: app/src/main/java/com/dancc/basketballstatskeeper/adapter/PlayerListAdapter.java
package com.dancc.basketballstatskeeper.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.dancc.basketballstatskeeper.R;
import java.util.List;

public class PlayerListAdapter
    extends RecyclerView.Adapter<PlayerListAdapter.PlayerNameViewHolder> {

  private List<String> players;

  public PlayerListAdapter(List<String> players) {
    this.players = players;
  }

  @NonNull
  @Override
  public PlayerNameViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View v = LayoutInflater.from(parent.getContext())
        .inflate(R.layout.player_list_cell_view, parent, false);

    return new PlayerNameViewHolder(v);
  }

  @Override
  public void onBindViewHolder(@NonNull PlayerNameViewHolder holder, int position) {
    holder.bind(players.get(position));
  }

  @Override
  public int getItemCount() {
    return players.size();
  }

  static class PlayerNameViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.playerName)
    TextView playerName;


    PlayerNameViewHolder(View v) {
      super(v);

      ButterKnife.bind(this, v);
    }

    void bind(String name) {
      playerName.setText(name);
    }
  }
}

File: app/src/main/java/com/dancc/basketballstatskeeper/adapter/PlayerListAdapter.kt
package com.dancc.basketballstatskeeper.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.dancc.basketballstatskeeper.R.layout
import com.dancc.basketballstatskeeper.adapter.PlayerListAdapter.PlayerNameViewHolder
import kotlinx.android.synthetic.main.player_list_cell_view.view.playerName

class PlayerListAdapter : Adapter<PlayerNameViewHolder>() {
  private var players = mutableListOf<String>()

  fun setPlayers(players: List<String>) {
    this.players.clear()
    this.players.addAll(players)
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlayerNameViewHolder {
    val v = LayoutInflater.from(parent.context)
        .inflate(layout.player_list_cell_view, parent, false)
    return PlayerNameViewHolder(v)
  }

  override fun onBindViewHolder(holder: PlayerNameViewHolder, position: Int) {
    if (position != 0) {
      holder.bind(players[position - 1])
    }
  }

  override fun getItemCount(): Int {
    return players.size + 1
  }

  class PlayerNameViewHolder(v: View) : ViewHolder(v) {

    fun bind(name: String?) {
      itemView.playerName.text = name
    }
  }
}